solr/core/src/java/org/apache/solr/search/SolrIndexSearcher.java

```diff
@@ -335,7 +335,7 @@ public SolrIndexSearcher(
       boolean reserveDirectory,
       DirectoryFactory directoryFactory)
       throws IOException {
-    super(wrapReader(core, r));
+    super(wrapReader(core, r), core.getCoreContainer().getCollectorExecutor());
```

Comment:
This change here is not controlled by the request level `multiThreaded=true|false` flag. #2570 explores how node-level opt-out could be supported.